frontend/src/toolbar/elements/heatmapLogic.ts

```diff
@@ -149,14 +151,19 @@ export const heatmapLogic = kea<heatmapLogicType>([
 
     selectors(({ cache }) => ({
         elements: [
-            (selectors) => [selectors.elementStats, toolbarLogic.selectors.dataAttributes, selectors.pageElements],
-            (elementStats, dataAttributes, pageElements) => {
+            (selectors) => [
+                selectors.elementStats,
+                toolbarLogic.selectors.dataAttributes,
+                selectors.pageElements,
+                selectors.matchLinksByHref,
+            ],
+            (elementStats, dataAttributes, pageElements, matchLinksByHref) => {
                 const elements: CountedHTMLElement[] = []
                 elementStats?.results.forEach((event) => {
                     let combinedSelector: string
                     let lastSelector: string | undefined
                     for (let i = 0; i < event.elements.length; i++) {
-                        const selector = elementToSelector(event.elements[i], dataAttributes) || '*'
+                        const selector = elementToSelector(event.elements[i], dataAttributes, matchLinksByHref) || '*'
```

Comment:
What about `

```suggestion
                        const element = event.elements[i]
                        const selector = elementToSelector(matchLinksByHref ? element : { ...element, href: undefined }, dataAttributes) || '*'
```

That way the `elementToSelector` function doesn't need to know about the changes you want to make to the element.